### PR TITLE
Finding and testing lowest possible versions

### PR DESCRIPTION
**Pull request by renatomefi**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Documented?   | no
| License       | MIT

- Trying to find out the lowest possible versions the bundle is compatible with
- Slight improves on travis build

----- OLD
- Sadly we don't support expression language 3.1
- Check it's constructor: https://github.com/symfony/symfony/blob/v3.1.10/src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php#L38
- Now on version 3.2: https://github.com/symfony/symfony/blob/v3.2.0/src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php#L40

And that's how we use it, as on 3.2: https://github.com/overblog/GraphQLBundle/blob/v0.9.2/ExpressionLanguage/ExpressionLanguage.php#L26

### CODE REVIEW

**Comment by renatomefi** [comment]
This is going to be a bit more trick since some tests are trying to force 3.1

**Comment by mcg-web** [comment]
I don't understand why expression language 3.1 is not compatible?

**Comment by renatomefi** [comment]
Because the way we are constructing it on https://github.com/overblog/GraphQLBundle/blob/v0.9.2/ExpressionLanguage/ExpressionLanguage.php#L26

This constructor is only available on 3.2, not on 3.1

**Comment by mcg-web** [comment]
both  constructor are compatible (2 params) we using only the second parameter right now to register providers. If it was not compatible tests would break? I don't get it sorry

**Comment by renatomefi** [comment]
They don't break because currently we are not testing with "--prefer-lowest", and there's a BC trick happening, so we are able to use expression language 3.2 with symfony 3.1.

I found it out while testing --prefer-lowest on the GraphiQL.

If we start doing a prefer-lowest here we are gonna find some nasty symfony breaks and we might have to fix each version

**Comment by mcg-web** [comment]
We should maybe add the flag if this is the case?

**Comment by renatomefi** [comment]
For some reason here it didn't fail on the 3.1, while if I make a functional test on the new GraphiQL bundle it fails on 3.1 and passes on 3.2

File: composer.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
         "react/promise": "To use ReactPHP promise adapter"
     },
     "require-dev": {
-        "phpunit/phpunit": "^4.1 || ^5.5 || ^6.0",
+        "phpunit/phpunit": "^5.5 || ^6.0",
```
Comment (renatomefi) [inline composer.json]: Since we use `TestCase` and not `PHPUnit_Framework_TestCase` we aren't compatible with phpunit 4

**Comment by renatomefi** [comment]
@mcg-web I've added the prefer-lowest build, it all looks fine!
I squashed since the history wasn't adding value

**Pull request merged**